addon/lib/user.js

```diff
@@ -64,3 +68,129 @@ function makeUuid() {
 exports.getUserInfo = function () {
   return ss.storage.userInfo;
 };
+
+exports.updateProfileInfo = function (profile) {
+  if (! profile) {
+    return;
+  }
+  let currentInfo = ss.storage.profileInfo;
```

Comment:
since we have to set ss.storage.profileInfo anyway, this could be `let currentInfo = ss.storage.profileInfo || {}` and forget the next if statement